### PR TITLE
Fix stuck in loop while looking for new interfaces

### PR DESCRIPTION
**Pull request by pokgak**

### CODE REVIEW

File: lib/libndmgmt/iface.cpp [hunk 1]
```diff
@@ -972,11 +972,13 @@ void network_interface::look_for_new_interfaces(rpl_debug *debug)
 {
     network_interface *iface = network_interface::all_if;
     while(iface != NULL) {
+        network_interface *next = iface->next;
         if(!iface->alive && !iface->disabled) {
             iface->setup();
 
             dag_network::send_all_dag(iface);
         }
+        iface = next;
```
Comment (smlng) [inline lib/libndmgmt/iface.cpp]: you can simplify this and use `iface = iface->next;` here omitting the extra variable

**Review by smlng — CHANGES_REQUESTED** [review]
minor nit, otherwise fixes the issue we found

**Review by smlng — APPROVED** [review]

**Comment by pokgak** [comment]
Squashed.

**Comment by smlng** [comment]
please rebase on latest master to make Travis-CI happy

**Comment by pokgak** [comment]
Rebased on latest master. Checks all green now.

**Pull request merged**